lib/ConnectionContextBase.ts

```diff
@@ -131,8 +131,8 @@ export module ConnectionContextBase {
    * the base connection context.
    */
   export function create(parameters: CreateConnectionContextBaseParameters): ConnectionContextBase {
-    if (!parameters || typeof parameters !== "object") {
-      throw new Error("'parameters' is a required parameter and must be of type 'object'.");
+    if (!parameters) {
+      throw new TypeError("'parameters' is a required parameter.");
```

Comment:
Not sure I completely understand, the next line is checking `parameters.config` and not `parameters`.  This check makes sense to me because it is checking the presence of `parameters` before checking its properties